data/src/message.rs

```diff
@@ -47,13 +49,10 @@ impl Message {
         }
     }
 
-    pub fn nickname(&self) -> Option<&str> {
+    pub fn user(&self) -> Option<User> {
         match &self {
-            Message::Sent { nickname, .. } => Some(nickname.as_str()),
-            Message::Received(message) => message.prefix.as_ref().and_then(|prefix| match prefix {
-                irc::proto::Prefix::ServerName(_) => None,
-                irc::proto::Prefix::Nickname(nickname, _, _) => Some(nickname.as_str()),
-            }),
+            Message::Sent { nickname, .. } => Some(User::new(nickname)),
```

Comment:
I think we, down the line, need to do a `WHOIS` on ourself because we don't know much about yourself here. And in my case where im connected through ZNC my nickname is actually different from the config:

<img width="240" alt="Screenshot 2023-06-09 at 14 32 43" src="https://github.com/squidowl/halloy/assets/2248455/8e15b35c-18eb-4956-840c-41e25ed161d5">